### PR TITLE
expand on preprocessor section

### PR DESCRIPTION
**Pull request by WalterBright**
to reflect the greatly increased preprocessor support in ImportC.

### CODE REVIEW

File: spec/importc.dd [hunk 1]
```diff
@@ -101,6 +147,25 @@ $(H2 $(LNAME2 preprocessor, Preprocessor))
 
     $(P and the preprocessed output is written to $(TT file.i).)
 
+    $(H3 $(LNAME2 defines, Preprocessor Macros))
+
+    $(P ImportC collects all the $(TT #define) macros from the preprocessor run when it is run automatically.
+    The macros that look like manifist constants, such as:)
```
Comment (maxhaton) [inline spec/importc.dd:153]:
```suggestion
    The macros that look like manifest constants, such as:)
```

Comment (WalterBright) [reply spec/importc.dd]: done

**Comment by ibuclaw** [comment]
Is there a note at the top saying this is an implementation detail, and describes how preprocessing works for dmd only?

**Comment by WalterBright** [comment]
I don't know what the state of ImportC is with gcd or ldc. I'll leave that to @ibuclaw and @kinke to make adjustments to the documentation.

**Review by dkorpel — APPROVED** [review]

**Pull request merged**